modules/api/src/main/java/org/apache/ignite/lang/util/IgniteNameUtils.java

```diff
@@ -108,9 +108,32 @@ public static String quoteIfNeeded(String identifier) {
         return identifier;
     }
 
+    /**
+     * Returns {@code true} if given string is valid identifier, {@code false} otherwise.
+     */
+    public static boolean isValidIdentifier(String identifier) {
```

Comment:
May be it's worth add tests for this method.

I suggest add check to `malformedIdentifiers`
```
assertThat(IgniteNameUtils.isValidIdentifier(source), is(false));
```
and may be split test `validaIdentifiers` into two `validQuotedIdentifiers` `validUnquotedIdentifiers` :thinking: